cpython-unix/build-main.py

```diff
@@ -129,9 +129,26 @@ def main():
 
     musl = "musl" in target_triple
 
+    # Linux targets can be built on a macOS host using Docker.
+    building_linux_from_macos = sys.platform == "darwin" and "linux" in target_triple
+    if building_linux_from_macos:
+        print("Note: Using Docker to build for Linux on macOS")
+        args.no_docker = False
```

Comment:
This feels a little dubious as an implementation. I'd expect us not to mutate `args` and I'd expect `--no-docker` to cause a failure in this case?